Playbook for workshop organization (#220)
coderefinery/manuals: workshop_playbook.md

# Workshop playbook

This page can be seen as a master to-do of things that need to be considered when planning a CodeRefinery workshop.
It may also help with the organization of other workshops.

## Before the workshop

### Organization team

* Find dates that fit most people that indicated interest in the workshop; 
    * CodeRefinery Zulip chat is a good place to find those
    * Also ask from known supporters (people organizing in-person or online exercise rooms)
* Find people to fill core-roles of the organization team 
    * Registration coordinator
    * Instructor coordinator
    * Director/Broadcaster
    * Host

> * Decide if certification is possible and what is needed for getting a certificate/credits (ask from partner universities)

Reviewer: ```suggestion
* Decide if certificates will be possible and what is needed for getting a certificate/credits (ask from partner universities)
```

* Find volunteer (co-)instructors; Zulip and "known associates"; also try to recruit new instructors from other channels
* Distribute lessons
    * via planning HackMD
    * ask instructors to pick lessons, eg add their initials to lesson of interest
    * minimum 2 instructors per lesson
* remind instructors to also register for workshop
* keep instructor onboarding
    * remind to check issues and possibilities for adjustments, contributions of "their" lessons
    * introduce CodeRefinery way of teaching if needed
* set up and inform team about installation and exercise leader onboarding sessions
* test and, if needed update installation instructions

### Workshop page and other planning tools

- Set up a workshop page from template, and add it to "upcoming courses"
- Create planning HackMD

>     - provide all links to important pages there

Reviewer: ```suggestion
    - provide all links to important pages there
    - There are two HackMDs: one internal (but public) for organizers, one for workshop participants.  The second can be made later.
```
This is obvious and covered later, but maybe make it clear already?

- Put workshop schedule on separate page
- List all exercise leads (who consent to being listed) on the website
- List instructors on the website
- List expert helpers on the website

### Workshop tools

* create Zoom rooms and communicate their function and addresses
    * instructor Zoom
    * exercise Zoom
* Set up Q&A HackMD (+ archive) with nice URL

### Registration

* Create indico event for registration handling
    * Modify registration confirmation message to include links

> ### Advertizing

Reviewer: ```suggestion
### Advertising
```

- University staff (Computing, open sci, etc.) (people who can serve as local organizers and serve as local helpers)
- Potential instructors (teach, etc.)
- High-level management (provide us funding)

### Communication 

- Draft email templates early; collaborate
  - Observers (+ EL onboarding info)
  - unsure registrations
  - Exercise leads (+ onboarding info)
  - Stream learners
  - update indico auto-reply for last minute registrations
  - remind team contacts to update team info
- Calendar invites to EL/EH/instructors/organization team
- Exercise leader onboarding summary email

## Continuous / every few days:
- check indico and follow up on registrations
   - look at team registrations and check for problems
     - identify where we need to do something. e.g. persons who wanted an in-person room. or persons who were not sure but wanted to be informed. basically, "playing" with the registration data now is important to not forget anybody and not to forget follow-up
   - look at exercise leads and make sure they registered also for the onboarding
   - try to approve as early as possible
 - check Freshdesk for new support requests

## During the workshop

> - after every workshop day, send summary and preparation for next day email to all

Reviewer: thought: I think we should do more updating of the news on the website.  this can be done even during the day as the speakers mention things for the next day.
```suggestion
- during/after each workshop day, collect relevant news on the workshop webpage
- after every workshop day, send summary and preparation for next day email to all (can be based on the website news and link there so that it can be updated)
```

## After the workshop

- Port changes from workshop page to template page
- Check this page and update if necessary for future workshops
  - Document necessary steps and form rigid workflow
- Save detailed schedule in each lesson page and link to it from workshop page

>   -As the schedule and workshop content is quite established at this point can we include as much actual content to the workshop page temlpate repo? Maybe fork/copy it so we still preserve the blank temlpate for WSaaS

Reviewer: ```suggestion
  - As the schedule and workshop content is quite established at this point can we include as much actual content to the workshop page temlpate repo? Maybe fork/copy it so we still preserve the blank temlpate for WSaaS
```
